### DISCUSSION TITLE
Installed torch-version 2.0.1 unused in model.train (yolov8)??!

### DISCUSSION THREAD

**Discussion by elanI338** [ultralytics/ultralytics#3143] [Q&A]
Hello community,

i am having a strange anomaly when trying to train in my jupyter notebook using the following command:

`from ultralytics import YOLO`
`model = YOLO("yolov8x-seg_custom.yaml")`
`results = model.train(data="custom_data.yaml", epochs=1000, workers=1, batch=20, imgsz=640)  # train the model`

training works, but i get the misleading information:

> Ultralytics YOLOv8.0.116  Python-3.9.12 _**torch-1.13.1**_ CUDA:0 (NVIDIA GeForce RTX 3090 Ti, 24564MiB)
> WARNING  Upgrade to torch>=2.0.0 for deterministic training.

I don't get it. I use conda list to check my actual torch version and it cleary returns:

> _**torch  2.0.1**_ 

What can i do to fix this? When i try pip install and upgrade the torch, it says its already satisfied.

Thank you very much in advance!

**Comment by kalclark** [comment]
Add the --upgrade option to your pip3 torch install, should uninstall pytorch <2 and allows ultralytics to use new pytorch version

For example:
`pip3 install --upgrade torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118`

**Reply by glenn-jocher** [reply]
@kalclark hello!

It seems like there might be a discrepancy between the PyTorch version detected by YOLOv8 and the one installed in your environment. This can sometimes happen due to multiple environments or incorrect path settings.

First, ensure that your Jupyter notebook is running in the correct environment where PyTorch 2.0.1 is installed. You can check the PyTorch version within the notebook by running:

```python
import torch
print(torch.__version__)
```

If the version printed is not 2.0.1, you may need to adjust your environment or kernel settings in Jupyter.

If the correct version is displayed, but the warning persists, it could be an issue with the YOLOv8 package not recognizing the updated version. In this case, you might want to check for updates to the YOLOv8 package itself or report the issue on the GitHub repository so that the team can investigate further.

Remember, you can always refer to the documentation for more detailed guidance on managing your training environment and troubleshooting common issues. 😊
